runtime/Cpp/runtime/src/atn/ATNSimulator.cpp

```diff
@@ -23,12 +23,11 @@ void ATNSimulator::clearDFA() {
   throw UnsupportedOperationException("This ATN simulator does not support clearing the DFA.");
 }
 
-PredictionContextCache& ATNSimulator::getSharedContextCache() {
+PredictionContextCache& ATNSimulator::getSharedContextCache() const {
```

Comment:
Usually it is a good idea to _not_ make a method const if it returns a non-const reference, i.e. allows the caller to change the internal state of the object by accessing the writable reference to an internal member ( https://google.github.io/styleguide/cppguide.html#Use_of_const ).

Probably a conscious choice, but just want to caution.

Reply:
In this case, the PredictionContextCache is not actually owned by the ATNSimulator, its given a reference to it during construction.